crates/driver/src/infra/solver/protocol_fees.rs

```diff
@@ -0,0 +1,271 @@
+use {
```

Comment:
This struct encapsulates the protocol fee at `solver/dto` level. Implementing `apply_to_order()` and `apply_to_solution()` functions.